### PR TITLE
Visitor adds item to cart#26

### LINKED ISSUE #45: User Story 26, User adds an item to the cart
```
As a visitor or registered user
When I visit an item's show page from the items catalog
I see a link or button to add this item to my cart
And I click its link or button
I am returned to the item index page
I see a flash message indicating the item has been added to my cart
The navigation bar increments my cart counter
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #46

depends on #57

### PR DESCRIPTION
**Pull request by danbriechle**
Adds test for registered user adding item to cart
Adds test for visitor adding item to cart 

Edits path in item show 

tests passing 

closes #45

### CODE REVIEW

File: spec/features/item_show_page_spec.rb [hunk 1]
```diff
@@ -28,4 +28,89 @@
...
+
+      expect(current_path).to eq(items_path)
+
+      within '#nav' do
+
+        expect(page).to have_content("Cart: 1")
+
+      end
+
+      expect(page).to have_content(item_1.name)
```
Comment (asmolentzov) [inline spec/features/item_show_page_spec.rb]: Not sure that testing for the items index page to have all the items is really necessary here - we do that elsewhere right? Might be superfluous for this test.

Comment (danbriechle) [reply spec/features/item_show_page_spec.rb]: Noted and removed.
